pkg/operator/baremetal_pod.go

```diff
@@ -264,6 +294,44 @@ func checkForBaremetalClusterOperator(osClient osclientset.Interface) (bool, err
 	return true, nil
 }
 
+// we define this service in code instead of a yaml manifest because
+// we only want it to exist when metal3 is deployed
+func newMetal3Service(config *OperatorConfig, baremetalProvisioningConfig BaremetalProvisioningConfig) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "metal3-metrics",
+			Namespace: config.TargetNamespace,
+			Annotations: map[string]string{
+				maoOwnedAnnotation: "",
+				// copied from settings for machine-api-controllers
+				// service in
+				// 0000_30_machine-api-operator_10_service.yaml
+				"service.alpha.openshift.io/serving-cert-secret-name":    certStoreName,
+				"exclude.release.openshift.io/internal-openshift-hosted": "true",
+			},
+			Labels: map[string]string{
+				"k8s-app": "controller",
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeNodePort,
+			Ports: []corev1.ServicePort{
+				{
+					Name: "metal3-mtrc",
+					Port: metal3ExposeMetricsPort,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.String,
+						StrVal: "metal3-mtrc",
```

Comment:
This doesn't appear to be a real [IANA service type](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml). How does this work?